src/main/java/org/folio/marc/migrations/services/batch/mapping/MappingRecordsChunkPreProcessor.java

```diff
@@ -30,13 +33,20 @@ public class MappingRecordsChunkPreProcessor implements ItemProcessor<OperationC
 
   private final AuthorityJdbcService authorityJdbcService;
   private final ChunkStepJdbcService chunkStepJdbcService;
+  private final InstanceJdbcService instanceJdbcService;
+  private final OperationJdbcService operationJdbcService;
 
   @Override
   public MappingComposite<MarcRecord> process(OperationChunk chunk) {
     log.trace("process:: for operation {} chunk {}", chunk.getOperationId(), chunk.getId());
     var chunkStep = createChunkStep(chunk);
 
-    var records = authorityJdbcService.getAuthoritiesChunk(chunk.getStartRecordId(), chunk.getEndRecordId());
+    var operationType = operationJdbcService.getOperation(chunk.getOperationId().toString()).getEntityType();
```

Comment:
Isn't operation type present in spring batch job parameters?

Reply:
no, because OperationChunk only contains operationId not entityType, that's why additional call was added to operationJdbcService

Reply:
We pass operation object to `MigrationOrchestrator` and put operationId in spring batch job parameters on job start. Than it's accessed in `ItemReader` like this `@Value("#{jobParameters['operationId']}"` and in `SavingRecordsStepListener` using `implements StepExecutionListener` and `afterStep(StepExecution stepExecution)`.
Probably makes sense to add operation type to job parameters in `MigrationOrchestrator` to avoid additional database calls for each chunk in multiple processors.